Remove redundant getReference(IdentifierProcessing) method from SqlId…

Mikhail2048 opened the pull request:
Fixes #1110 issue. Jens @schauder, can you, please, review these changes?

schauder commented on spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/SqlIdentifier.java:
```diff
@@ -59,13 +60,13 @@ public SqlIdentifier transform(UnaryOperator<String> transformationFunction) {
 		@Override
-		public String toSql(IdentifierProcessing processing) {
-			throw new UnsupportedOperationException("An empty SqlIdentifier can't be used in to create SQL snippets");
+		public String getReference() {
```
Please do not change the order of methods.

It makes reading the diffs unnecessary hard.

Mikhail2048 replied:
Yeah, sure, I have fixed this with [this](https://github.com/spring-projects/spring-data-jdbc/pull/1111/files#diff-ad9a19ceaf2639443aa86e26bc0cd97e5e9b2aad5ea158cae9477246b5d38294L79) commit

schauder commented on spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/SqlIdentifier.java:
```diff
@@ -74,24 +75,13 @@ public String toString() {
-
-	/**
-	 * Return the reference name without any further transformation. The reference name is used for programmatic access to
+	 * Return the name without any further transformation. The reference name is used for programmatic access to
```
We shouldn't directly remove public methods. Instead please deprecate it and all it's implementations. We then should remove it with 3.0

Mikhail2048 replied:
Fixed with [commit](https://github.com/spring-projects/spring-data-jdbc/pull/1111/files#diff-ad9a19ceaf2639443aa86e26bc0cd97e5e9b2aad5ea158cae9477246b5d38294L82)

schauder reviewed (changes requested):
Nice catch.

I have just two minor changes to request.

Mikhail2048 commented:
@schauder I have addressed the issues you have mentioned. Please, rereview the changes again. I have cleaned up all usages of `SqlIndentifier#getReference(IdentifierProcessing)` method within the project, but left the actual method and its implementations in place (marked as `@Deprecated`) in order for it to be deleted in 3.0.0 release

Mikhail2048 commented:
Hi Jens @schauder! Are there any news about this PR?)
